src/Symfony/Component/Console/Input/InputOption.php

```diff
@@ -177,7 +177,7 @@ public function isNegatable(): bool
      */
     public function setDefault($default = null)
     {
-        if (self::VALUE_NONE === (self::VALUE_NONE & $this->mode) && null !== $default) {
+        if (self::VALUE_NONE === (self::VALUE_NONE & $this->mode) && !$this->isNegatable() && null !== $default) {
```

Comment:
If `VALUE_NEGATABLE` means `VALUE_NONE`, we can force `$this->mode |= self::VALUE_NODE` when `VALUE_NEGATABLE` is detected. That would also throw an error when `VALUE_NEGATABLE | VALUE_OPTIONAL` is given (with your proposition in #52055).

Reply:
The first bit is already covered in #52055: if `VALUE_REQUIRED` or `VALUE_OPTIONAL` is not provided, then `VALUE_NONE` is added by default:
https://github.com/symfony/symfony/blob/251b2424bc234ac0676db53f398194da1c822fbb/src/Symfony/Component/Console/Input/InputOption.php#L95-L96
So #52055 makes `VALUE_NEGATABLE | VALUE_NONE` and (just) `VALUE_NEGATABLE` behave in exactly the same manner, which was the reason for me finding this bug.

The error on `VALUE_NEGATABLE | VALUE_OPTIONAL` (or `VALUE_REQUIRED`) already exists:
https://github.com/symfony/symfony/blob/ba2f4a5d9cbab20945c60d57e570fe3004ca1c51/src/Symfony/Component/Console/Input/InputOption.php#L103-L105